frontend/packages/core/src/Contexts/app-context.tsx

```diff
@@ -2,9 +2,21 @@ import React from "react";
 
 import type { Workflow } from "../AppProvider/workflow";
 
+export type HeaderItems = "NPS";
+
+export interface TriggeredHeaderData {
+  [key: string]: {
+    open: boolean;
+    [key: string]: unknown;
+  };
+}
```

Comment:
Currently made this very generic so we could support other options easily. However I can see a case for it being strongly typed as we're unlikely to have a large amount of items in the Header so I'm open to opinions

Reply:
Added a comment and an example to indicate usability